lib/services/saml.go

```diff
@@ -103,6 +103,11 @@ func ValidateSAMLConnector(sc types.SAMLConnector, rg RoleGetter) error {
 	if rg != nil {
 		for _, mapping := range sc.GetAttributesToRoles() {
 			for _, role := range mapping.Roles {
+				if utils.ContainsExpansion(role) {
+					// Role is a template so we cannot check for existence of that literal name.
+					continue
+				}
+				role = strings.ReplaceAll(role, "$$", "$")
```

Comment:
Let's please add a comment here explaining why this is needed, and also verify this behavior on a real setup with real roles.

Reply:
Turns out not to be needed as a role name cannot have a `$` in it, so I've removed this replacement.